cljfmt/test/cljfmt/core_test.cljc

```diff
@@ -4,289 +4,750 @@
+          ""]))
+    (is (reformats-to?
+         ["(foo"
+          "(bar"
+          ")"
+          ")"]
+         {:remove-surrounding-whitespace? false}
+         ["(foo"
+          " (bar"
+          "  )"
+          " )"])
+        "indents properly")))
 
 (deftest test-options
-  (is (= (reformat-string "(foo)\n\n\n(bar)" {:remove-consecutive-blank-lines? false})
-         "(foo)\n\n\n(bar)"))
-  (is (= (reformat-string "(  foo  )" {:remove-surrounding-whitespace? false})
-         "(  foo  )"))
-  (is (= (reformat-string "(foo(bar))" {:insert-missing-whitespace? false})
-         "(foo(bar))"))
-  (is (= (reformat-string "(foo\nbar)" {:indents '{foo [[:block 0]]}})
-         "(foo\n  bar)"))
-  (is (= (reformat-string "(do\nfoo\nbar)" {:indents {}})
-         "(do\n foo\n bar)"))
-  (is (= (reformat-string "(do\nfoo\nbar)" {:indentation? false})
-         "(do\nfoo\nbar)"))
-  (is (= (reformat-string "(foo bar) \n(foo baz)" {:remove-trailing-whitespace? false})
-         "(foo bar) \n(foo baz)"))
-  (is (= (reformat-string "(foo bar) \n" {:remove-trailing-whitespace? false})
-         "(foo bar) \n")))
+  (is (reformats-to?
+       ["(foo)"
+        ""
+        ""
+        "(bar)"]
+       {:remove-consecutive-blank-lines? false}
+       ["(foo)"
+        ""
+        ""
+        "(bar)"]))
+  (is (reformats-to?
+       ["(  foo  )"]
+       {:remove-surrounding-whitespace? false}
+       ["(  foo  )"]))
+  (is (reformats-to?
+       ["(foo(bar))"]
+       {:insert-missing-whitespace? false}
+       ["(foo(bar))"]))
+  (is (reformats-to?
+       ["(foo"
+        "bar)"]
+       {:indents {'foo [[:block 0]]}}
+       ["(foo"
+        "  bar)"])
+      "can customize block indentation")
+  (is (reformats-to?
+       ["(do"
+        "foo"
+        "bar)"]
+       {:indents {}}
+       ["(do"
+        " foo"
+        " bar)"])
+      "can clear all indents rules")
+  (is (reformats-to?
+       ["(do"
+        "foo"
+        "bar)"]
+       {:indentation? false}
+       ["(do"
+        "foo"
+        "bar)"])
+      "can disable indentation")
+  (is (reformats-to?
+       ["(foo bar) "
+        "(foo baz)"]
+       {:remove-trailing-whitespace? false}
+       ["(foo bar) "
+        "(foo baz)"]))
+  (is (reformats-to?
+       ["(foo bar) "
+        ""]
+       {:remove-trailing-whitespace? false}
+       ["(foo bar) "
+        ""])))
 
 (deftest test-parsing
-  (is (= (reformat-string ";foo") ";foo"))
-  (is (= (reformat-string "::foo") "::foo"))
-  (is (= (reformat-string "::foo/bar") "::foo/bar"))
-  (is (= (reformat-string "foo:bar") "foo:bar"))
-  (is (= (reformat-string "#_(foo\nbar)") "#_(foo\n   bar)"))
-  (is (= (reformat-string "(juxt +' -')") "(juxt +' -')"))
-  (is (= (reformat-string "#\"(?i)foo\"") "#\"(?i)foo\""))
-  (is (= (reformat-string "#\"a\nb\"") "#\"a\nb\"")))
+  (is (reformats-to?
+       [";foo"]
+       [";foo"]))
+  (is (reformats-to?
+       ["::foo"]
+       ["::foo"])
+      "auto-resolved keyword")
```

Comment:
I don't think we need messages where the reformatting is obvious or trivial.

Reply:
I am certainly a newbie in some areas of Clojure, if I looked it up to double check what it is, I probably added a msg.  I'll review.